File: songs/lua/07-flatten.lua
local inspect = require('inspect')
require 'my-songs'

function flatten(songs)
  local tags = {}

  for _, song in pairs(songs)
  do
    local song_tags = song.tags

    if type(song_tags) == 'table' then
      for _, tag in pairs(song_tags)
      do
        tags[#tags + 1] = tag
      end
    end

  end
  
  return tags
end

print(inspect(flatten(songs)))

File: songs/ruby/07-flatten.rb
require_relative 'MySongs'
include Songs

def flatten(songs)
  tags = []
  
  songs.each { |song| 
    if song['tags'] != nil
       song['tags'].each { |tag| tags << tag }
   end  
  }
  
  tags
end

puts "#{flatten(SONGS)}"
